cmd/gotosocial/action/admin/media/list.go

```
@@ -0,0 +1,80 @@
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package media
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/superseriousbusiness/gotosocial/cmd/gotosocial/action"
+	"github.com/superseriousbusiness/gotosocial/internal/config"
+	"github.com/superseriousbusiness/gotosocial/internal/db/bundb"
+	"github.com/superseriousbusiness/gotosocial/internal/state"
+)
+
+var List action.GTSAction = func(ctx context.Context) error {
+	remote := config.GetAdminMediaListRemote()
+	local := config.GetAdminMediaListLocal()
+
+	if !local && !remote {
+		return fmt.Errorf("need to pass at least one of --%s, --%s", config.AdminMediaListRemoteFlag(), config.AdminMediaListLocalFlag())
+	}
+
+	var state state.State
+	state.Caches.Init()
+	state.Workers.Start()
+
+	dbConn, err := bundb.NewBunDBService(ctx, &state)
+	if err != nil {
+		return fmt.Errorf("error creating dbservice: %s", err)
+	}
+
+	// Set the state DB connection
+	state.DB = dbConn
+
+	attachments, err := dbConn.GetAttachments(ctx, "", 0)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve attachments: %w", err)
+	}
+
+	f := bufio.NewWriter(os.Stdout)
+
+	mediaPath := config.GetStorageLocalBasePath()
+
+	for _, a := range attachments {
+		if local && a.RemoteURL == "" {
+			if _, err := f.WriteString(path.Join(mediaPath, a.File.Path) + "\n"); err != nil {
```

Not sure if we should return the error here, or just ignore it entirely. Likelihood of this causing an error is essentially 0, unless `os.Stdout` gets banged up somehow. It does have the advantage that if we do run into an issue, the command will exit non-zero so it's possible to detect that generating the list experienced an error. That seems desirable as the output of this command is meant to feed into other systems.

---

honestly i would just drop the error here. a stdout error indicates a serious issue that is going to stop this entire process from working. chances are stderr would be fucked too which is why i wouldn't bother with a panic either.

---

also makes the code here much more concise :)